### PR TITLE
Add samples for go and nodeJs, fix readme

### PR DESCRIPTION
**Pull request by kporwal-aws**
*Issue #, if available:*

*Description of changes:* Adding sample application for Go and NodeJS.  Add region as input argument for java apps and fix readme.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sample_apps_reinvent2021/go/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  ```
+
+- Run example to create scheduled query that fails on execution and generates an error report. DEFAULT value is false,
+  so it runs a valid query in the scheduled query sample
+  ```
+  go run scheduled-query-sample.go --run_invalid_scheduled_query_flag=true
+  ```
+- Delete resources created by this script for creating/executing E2E scheduled query. DEFAULT value is true
+  ```
+  go run scheduled-query-sample.go --delete_resources_after_execution_flag=false
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/README.md]: should the value be true or false. If false, fix the description

File: sample_apps_reinvent2021/go/crud-ingestion-sample.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	sess, err := utils.GetSession(*region)
+	utils.HandleError(err, "Failed to start a new session", true)
+
+	writeSvc := timestreamwrite.New(sess)
+	querySvc := timestreamquery.New(sess)
+	s3Svc := s3.New(sess, aws.NewConfig().WithRegion(*region))
+
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Println("Creating a database, hit enter to continue")
+	reader.ReadString('\n')
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/crud-ingestion-sample.go]: Why do we need user input after each step?

File: sample_apps_reinvent2021/go/ingestion-csv-sample.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	//Ingest records from csv file
+	timestreamBuilder.IngestRecordsFromCsv(*testFileName, *databaseName, *tableName)
+
+	// sample query
+	queryString := fmt.Sprintf("select count(*) from %s.%s", *databaseName, *tableName)
+	// execute the query
+	queryOutput, err := timestreamBuilder.QueryWithQueryString(queryString)
+	if err == nil {
+		fmt.Println(queryOutput)
+	}
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/ingestion-csv-sample.go:99]: Not cleaning up resources?

File: sample_apps_reinvent2021/go/devops.sh [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+    GROUP BY hostname
+)
+SELECT REDUCE(interpolated_cpu_utilization,
+    CAST(ROW(0.0, 0) AS ROW(sum DOUBLE, count INTEGER)),
+    (s, x) -> CAST(ROW(x.value + s.sum, s.count + 1) AS ROW(sum DOUBLE, count INTEGER)),
+     s -> IF(s.count = 0, NULL, s.sum / s.count)) AS avg_cpu
+FROM time_series_view"
+
+# Try other functionality in query
+go run query-operations.go -query "SELECT * FROM $1.$2"
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/devops.sh]: Can we rename the script as query-with-cancel.go? Also move out the limit query to this script.

File: sample_apps_reinvent2021/go/query-operations.go [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	"go_sample/utils"
+
+	"flag"
+	"fmt"
+	"os"
+)
+
+func main() {
+	// setup the query client
+	sess, err := session.NewSession(&aws.Config{Region: aws.String("us-east-1")})
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/query-operations.go]: What about other regions?

File: sample_apps_reinvent2021/go/query-sample.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	"go_sample/utils"
+
+	"flag"
+	"fmt"
+	"os"
+)
+
+func main() {
+	// setup the query client
+	sess, err := session.NewSession(&aws.Config{Region: aws.String("us-east-1")})
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/query-sample.go]: what about other regions

File: sample_apps_reinvent2021/go/scheduled-query-sample.go [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+
+	sess, err := utils.GetSession(*region)
+	utils.HandleError(err, "Failed to start a new session", true)
+
+	utils.HandleError(err, "Failed to start a new session", true)
+
+	writeSvc := timestreamwrite.New(sess)
+	querySvc := timestreamquery.New(sess)
+
+	snsSvc := sns.New(sess)
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/scheduled-query-sample.go:54]: what about different regions?

File: sample_apps_reinvent2021/go/scheduled-query-sample.go [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+	utils.HandleError(err, fmt.Sprintf("Failed to describe scheduled query with scheduledQueryArn %s ", scheduledQueryArn), false)
+
+	err = timestreamBuilder.ExecuteScheduledQuery(scheduledQueryArn, time.Now())
+	utils.HandleError(err, fmt.Sprintf("Failed to execute scheduled query with scheduledQueryArn %s ", scheduledQueryArn), false)
+
+	err = timestreamBuilder.DescribeScheduledQuery(scheduledQueryArn)
+	utils.HandleError(err, fmt.Sprintf("Failed to describe scheduled query with scheduledQueryArn %s ", scheduledQueryArn), false)
+
+	// Sleep for 65 seconds to let ScheduledQuery run
+	fmt.Printf("Waiting for automatic ScheduledQuery executions & notifications from queueUrl %s", queueUrl)
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/scheduled-query-sample.go]: Include 65 seconds in the waiting message. Else cx will ctrl-c thinking that the app is hung

File: sample_apps_reinvent2021/go/scheduled-query-sample.go [hunk 9]
```diff
@@ -0,0 +1,261 @@
...
+
+		if topicArn != "" {
+			timestreamDependencyHelper.DeleteSnsTopic(topicArn)
+		}
+
+		timestreamBuilder.DeleteTable(sqResultsDatabaseName, sqResultsTableName)
+		timestreamBuilder.DeleteDatabase(sqResultsDatabaseName)
+		timestreamBuilder.DeleteTable(databaseName, tableName)
+		timestreamBuilder.DeleteDatabase(databaseName)
+	}
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/scheduled-query-sample.go:276]: can we print a message like 'Please delete the following resources' and list the arns of the all the resources in an else block

File: sample_apps_reinvent2021/go/utils/timestream-dependency-helper.go [hunk 10]
```diff
@@ -0,0 +1,625 @@
...
+	setQueueAttributesOutput, err := timestreamDependencyHelper.SqsSvc.SetQueueAttributes(setQueueAttributesInput)
+	if err != nil {
+		fmt.Printf("Error: %s\n", err.Error())
+	} else {
+		fmt.Printf("setQueueAttributesOutput is successful : %s", setQueueAttributesOutput.GoString())
+	}
+	return err
+}
+
+func (timestreamDependencyHelper TimestreamDependencyHelper) getIamRole(roleName string) (string, error) {
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/go/utils/timestream-dependency-helper.go]: GetIamRole

File: sample_apps_reinvent2021/js/main.js [hunk 11]
```diff
@@ -0,0 +1,237 @@
...
+        //Waiting for newly created role to be active
+        await new Promise(resolve => setTimeout(resolve, 15000));
+
+        var scheduledQueryArn = await scheduleQueryExample.createScheduledQuery(topicArn, roleArn, s3ErrorReportBucketName,constants.SQ_DATABASE_NAME, constants.SQ_TABLE_NAME);
+        //var scheduledQueryArn = await scheduleQueryExample.createInvalidScheduleQuery(topicArn, roleArn, s3ErrorReportBucketName, constants.SQ_DATABASE_NAME, constants.SQ_TABLE_NAME);
+
+        await scheduleQueryExample.listScheduledQueries();
+        await scheduleQueryExample.describeScheduledQuery(scheduledQueryArn);
+
+        console.log("Waiting for automatic ScheduledQuery executions & notifications");
```
Comment (rajeiyer) [inline sample_apps_reinvent2021/js/main.js]: Include 65 seconds in the message

File: sample_apps_reinvent2021/go/README.md (hunk 1 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/README.md]: we have set the default as true in the src.   also given we have option to pass a flag we should be good with any default. thoughts ?

File: sample_apps_reinvent2021/go/crud-ingestion-sample.go (hunk 2 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/crud-ingestion-sample.go]: I see. so this is similar to current application.   but I agree we do not need explicit customer input every time.

File: sample_apps_reinvent2021/go/ingestion-csv-sample.go (hunk 3 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/ingestion-csv-sample.go:99]: yes, again similar to current sample_app.   let me get back to you on why this decision is made.

File: sample_apps_reinvent2021/go/query-operations.go (hunk 5 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/query-operations.go]: so currently the samples seems to provide specific region input for ScheduledQuery samples only and the older samples are copied verbatim.    I agree we should make this change - let me take this up.

File: sample_apps_reinvent2021/go/query-sample.go (hunk 6 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/query-sample.go]: replied above.

File: sample_apps_reinvent2021/go/scheduled-query-sample.go (hunk 7 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/scheduled-query-sample.go:54]: we already do `sess, err := utils.GetSession(*region)` on L48 I believe.  Am I missing something here?

File: sample_apps_reinvent2021/go/scheduled-query-sample.go (hunk 8 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/scheduled-query-sample.go]: Ack,

File: sample_apps_reinvent2021/go/scheduled-query-sample.go (hunk 9 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/scheduled-query-sample.go:276]: hmm good idea.   let me do this for Go and later we can make it for other languages too.

File: sample_apps_reinvent2021/go/utils/timestream-dependency-helper.go (hunk 10 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/utils/timestream-dependency-helper.go]: Ack

File: sample_apps_reinvent2021/js/main.js (hunk 11 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/js/main.js]: ack

File: sample_apps_reinvent2021/go/devops.sh (hunk 4 above)
Comment (kporwal-aws) [reply sample_apps_reinvent2021/go/devops.sh]: Its a good suggestion but lets take this refactoring later as part of upcoming changes which are planned.

File: sample_apps_reinvent2021/go/README.md (hunk 1 above)
Comment (rajeiyer) [reply sample_apps_reinvent2021/go/README.md]: The comment says delete resources but flag value is false.

**Review by sethusrinivasan — APPROVED** [review]

**Review by rajeiyer — APPROVED** [review]

**Pull request merged**